sdk/bootstrap/ip.go

```diff
@@ -6,38 +6,42 @@ import (
 	"os"
 )
 
-// This helper function gives the current container's IP address, when run in a
-// given network, mnt and UTS namespace.  The function first checks the presence of
-// `LIBPROCESS_IP` if present it will return this as the container's IP address
-// else it will look at the `hostname` setup for the container to pick out the
-// IP address associated with the hostname.
-func ContainerIP() (containerIP net.IP, err error) {
-	ip := os.Getenv("LIBPROCESS_IP")
-	if ip != "" {
-		containerIP = net.ParseIP(ip)
-		if containerIP == nil {
-			err = fmt.Errorf("Invalid LIBPROCESS_IP found: %s", ip)
-			return
-		}
-
-		if containerIP.String() != "0.0.0.0" && containerIP.String() != "::" {
-			return
-		}
+var libprocessIP = "LIBPROCESS_IP"
+var mesosContainerIP = "MESOS_CONTAINER_IP"
+
+// ContainerIP This helper function gives the current container's IP address, when run in a
+// given network, mnt and UTS namespace.
+func ContainerIP() (net.IP, error) {
+	// First, try MESOS_CONTAINER_IP (DC/OS 1.10+, when running the default executor)
+	ip, err := validateIP(mesosContainerIP)
```

Comment:
Just a thought, might be easier to write:
envVar := []string{"MESOS_CONTAINER_IP", "LIBPROCESS_IP"}
for var in range(envVar) {
   ip, err := {
     //validate retrieval of IP based on the envVar.
   }()
}

Reply:
Not a bad idea. Didn't want to re-roll CI.